src/boot/efi/stub.c

```diff
@@ -277,6 +278,16 @@ static EFI_STATUS run(EFI_HANDLE image) {
                 mangle_stub_cmdline(cmdline);
         }
 
+        const char *extra = smbios_find_oem_string("io.systemd.boot.kernel-cmdline-extra");
+        if (extra) {
+                _cleanup_free_ char16_t *extra16 = xstr8_to_16(extra);
+                size_t sz = strlen16(cmdline);
+
+                cmdline = xrealloc(cmdline, (sz + 2) * sizeof(uint16_t), (sz + strlen8(extra) + 2) * sizeof(uint16_t));
+                cmdline[sz] = ' ';
```

Comment:
time to add strextend16()?

Reply:
We just use `xasprintf("%s %s")` everywhere.